src/main/java/chess/domain/piece/Pawn.java

```diff
@@ -0,0 +1,73 @@
+package chess.domain.piece;
+
+import chess.domain.board.position.Horizontal;
+import chess.domain.board.position.Position;
+import chess.domain.piece.movement.Direction;
+import chess.domain.piece.movement.Distance;
+
+import java.util.List;
+
+public abstract class Pawn extends Piece {
+    private static final Distance DIAGONAL_DISTANCE_WHEN_EXIST_ENEMY = Distance.ONE;
+    private static final Distance STRAIGHT_DISTANCE_WHEN_ON_FIRST_LINE = Distance.TWO;
+    private static final Distance STRAIGHT_DISTANCE = Distance.ONE;
+
+    private static final Pawn BLACK_PAWN = new Pawn(Owner.BLACK, Direction.blackPawnDirections()) {
+        @Override
+        public boolean isFirstLine(final Horizontal horizontal) {
+            return horizontal.equals(Horizontal.SEVEN);
+        }
+    };
+
+    private static final Pawn WHITE_PAWN = new Pawn(Owner.WHITE, Direction.whitePawnDirections()) {
+        @Override
+        public boolean isFirstLine(final Horizontal horizontal) {
+            return horizontal.equals(Horizontal.TWO);
+        }
+    };
+
+    private Pawn(final Owner owner, final List<Direction> directions) {
+        super(
+                owner,
+                Score.PAWN,
+                directions,
+                Distance.TWO
+        );
+    }
+
+    public static Pawn getInstanceOf(final Owner owner) {
+        if (owner.equals(Owner.BLACK)) {
```

Comment:
동치 비교 시에는 NPE를 방지하기 위해 항상 더 확실한 쪽을 기준으로 비교하는 것이 좋습니다.
```suggestion
        if (Owner.BLACK.equals(owner)) {
```

Reply:
```java
public static Pawn getInstanceOf(final Owner owner) {
    if (owner.isBlack()) {
        return BLACK_PAWN;
    }

    if (owner.isWhite()) {
        return WHITE_PAWN;
    }

    throw new IllegalArgumentException("Invalid pawn");
}
```
이렇게 바꿨습니다.

피드백의 내용은 이전 미션에서 피드백을 받았던 부분인데 아직 습관이 덜 되었던 것 같습니다.
NPE를 피하기 위해 알고 있는 (실천하지 못하고 있는) 나름의 규칙을 정리해보았습니다.
[블로그 / NPE를 피하기 위한 습관 정리](https://ecsimsw.tistory.com/entry/NullPointerException을-피하기-위한-습관)